discovery/file/file.go

```diff
@@ -73,6 +142,7 @@ func NewDiscovery(conf *config.FileSDConfig, logger log.Logger) *Discovery {
 	if logger == nil {
 		logger = log.NewNopLogger()
 	}
+	fileSDTimeStamp.logger = logger
```

Comment:
A constructor shouldn't really be setting a field on a global. I don't think it should be doing any logging.

Reply:
I left this in as I thought we might want to have logging if for some reason we can't access the file to get the timestamp, but that should be caught by this block in `refresh`:
```
tgroups, err := readFile(p)
		if err != nil {
			fileSDReadErrorsCount.Inc()

			level.Error(d.logger).Log("msg", "Error reading file", "path", p, "err", err)
			// Prevent deletion down below.
			ref[p] = d.lastRefresh[p]
			continue
		}
```